docs/guides/lint.md

```diff
@@ -3,75 +3,235 @@ redirectFrom:
+
+### Preprocessors
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+Preprocessors run first during `lint` and `bundle`.
+
+:::info
+Linting happens only when the `--lint` flag is supplied
+:::
+
+:::info
+Preprocessors are rarely indicated - avoid if possible.
+
+This section can be omitted.
+:::
+
+### Rules
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+For `lint` command: rules run *after* preprocessors.
+For `bundle` command: rules run *between* preprocessors and decorators.
+
+#### Examples
 
-  preprocessors:
-    # Preprocessors are rarely indicated -- avoid if possible.
-    # This section can be omitted.
+```yaml short version
+lint:
+  rules:
+    no-sibling-refs: error
+```
 
+```yaml verbose version
+lint:
   rules:
-    # Override any rules that are available from the extends configuration.
-    # Severity options are "error", "warn" or "off".
-    no-sibling-refs: error # Uses syntactic sugar to define the severity level of an option.
-    # Alternative verbose configuration
     no-sibling-refs:
       severity: error
-    # Some rules may have additional configuration options. Use the verbose configuration style in those cases.
-    # This boolean parameter prefixes example overrides the default "prefixes".
+```
+
+```yaml rules with additional configuration
+# Use verbose configuration version to define additional configuration
+# The boolean-parameter-prefixes example overrides the default "prefixes".
+lint:
+  rules:
     boolean-parameter-prefixes:
       severity: error
       prefixes: ['should', 'is', 'has']
+```    
 
-  decorators:
-    # Decorators modify the the definition after validation is complete, only in the bundling process.
-    # This section can be omitted if you don't use decorators.
-  ...
-```
+### Decorators
 
+Use this section to enable or disable decorators. They modify the definition in the bundling process after validation is complete.
 
-## Severity levels
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
 
-This applies to the preprocessors, rules, and decorators subsections of the lint configuration.
+For `bundle` command: decorators run *after* linting.
 
-This file excerpt shows utilizing syntactic sugar to control the problem severity.
+:::info
+Linting happens only when the `--lint` flag is supplied
+:::
 
-The severity level options are:
-- off
-- warn
-- error
+:::success
+Omit this section if you don't use decorators.
```

Comment:
Same suggestion as in previous comments by @swapnilogale - let's move this further up in the text (and consider taking it out of the admonition box)